### PR TITLE
HTMLalyzer

### PR DESCRIPTION
**Pull request by EdgesFTW**
A web interface all wrapped up into a single html file. This allows users to simply point a browser to a file to have access to a GUI that can analyze multiple output files of logpyle at once! 

image

image

### CODE REVIEW

File: bin/htmlalyzer [hunk 1]
```diff
@@ -16,6 +18,55 @@ def main() -> None:
...
+    # calculate hashes of files used to build html
+    hashes_str = ""
+
+    # get logpyle
+    with open(html_path+"/../__init__.py", "rb") as f:
+        binary_data = f.read()
+        m = hashlib.sha256()
+        m.update(binary_data)
+        hash = m.digest()
+        hashes_str += "logpyle:" + base64.b64encode(hash).decode("utf-8") + "\n"
```
[the remark this answers is not in the document]
Comment (matthiasdiener) [reply bin/htmlalyzer:30]: Also, you could put this function into `logpyle/HTMLalyzer/__init__.py` and import it here

[the remark this answers is not in the document]
Comment (EdgesFTW) [reply bin/htmlalyzer]: We now iterate over a list of filenames in the function get_current_hash()

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -187,3 +187,23 @@ jobs:
...
+        - uses: actions/checkout@v3
+        -
+            uses: actions/setup-python@v4
+            with:
+                python-version: '3.x'
+
+        - name: Run
+          run: |
+            pip install -e .
+            python -m pip install pymbolic psutil matplotlib Jinja2
```
Comment (matthiasdiener) [inline .github/workflows/ci.yaml:203]: Can Jinja be removed?

File: logpyle/HTMLalyzer/__init__.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            binary_data = f.read()
+            m = hashlib.sha256()
+            m.update(binary_data)
+            hash = m.digest()
+            hashes_str += file + ": " + base64.b64encode(hash).decode("utf-8") + "\n"
+
+    return hashes_str
+
+
+def setup() -> None:
```
Comment (matthiasdiener) [inline logpyle/HTMLalyzer/__init__.py:30]:
```suggestion
def build() -> None:
```

File: logpyle/HTMLalyzer/web-interface.html [hunk 4]
```diff
@@ -0,0 +1,694 @@
+<html>
```
Comment (matthiasdiener) [inline logpyle/HTMLalyzer/web-interface.html:1]: Maybe rename it to `htmlalizer.html`.

**Review by matthiasdiener — APPROVED** [review]
LGTM, thanks!

**Pull request merged**